### PR TITLE
performance changes for crash search

### PR DESCRIPTION
**Pull request by KarateCowboy**

### CODE REVIEW

File: src/models/fti.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// const pgd = require('pg')
+
+let FIND_BY_OBJECT_ID = `SELECT * FROM dtl.fti WHERE object_id = $1 LIMIT 1`
+let INSERT_INTO_ARCHIVE = `INSERT INTO dtl.fti_archive (id,object_type, object_id, searchable) VALUES($1,$2.$3,$4)`
```
Comment (aekeus) [inline src/models/fti.js]: Typo between $2 and $3

File: src/models/fti.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+let DELETE_BY_OBJECT_ID = `DELETE FROM dtl.fti WHERE object_id = $1`
+
+class FTI {
+
+  static async archive (object_id) {
+    // const pg = await pgd.connect(process.env.DATABASE_URL)
+    try {
+      const fti = await pg.query(FIND_BY_OBJECT_ID, object_id).rows[0]
```
Comment (aekeus) [inline src/models/fti.js]: object_id should be [object_id]

File: src/models/crashes.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+const moment = require('moment')
+const FTI = require('./fti')
+
+class Crashes {
+
+  static async archive (months=1) {
+    const cutoff = moment().subtract(months, 'months').format('YYYY-MM-DD')
+
+
+   const GET_OLD_CRASHES = `SELECT id, ts, contents, github_repo, github_issue_number FROM dtl.crashes WHERE ts < '${cutoff}' LIMIT 10000`
```
Comment (aekeus) [inline src/models/crashes.js]: ${cutoff} should be a parameter to a query call

Comment (aekeus) [reply src/models/crashes.js]: nit: spacing

File: src/models/crashes.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    let results = await pg.query(GET_OLD_CRASHES)
+    let transferred = 0
+    let keepProcessing = true
+    while(keepProcessing){
+      for (let row of results.rows) {
+        const INSERT = `INSERT INTO dtl.crashes_archive (id, ts, contents, github_repo, github_issue_number) VALUES 
+        ($1,$2,$3,$4,$5)`
+        try {
+          const values = [row.id, row.ts, row.contents, row.github_repo, row.github_issue_numebr]
+          await pg.query(INSERT,values)
```
Comment (aekeus) [inline src/models/crashes.js]: Where is the pg var defined and connected? Does it come from the script?

**Comment by KarateCowboy** [comment]
I made the requested changes.

I defined `pg` as a global in the script.

**Pull request merged**